File: python/1903.py
# def largestOddNumber(num: str) -> str:  # parsing every char
#     for i in range(len(num) - 1, -1, -1):
#         if int(num[i]) % 2 == 1:
#             return num[:i + 1]
#     return ""


# def largestOddNumber(num: str) -> str:
#     for i in range(len(num) - 1, -1, -1):
#         if num[i] in ["1", "3", "5", "7", "9"]:
#             return num[0: i + 1]
#     return ""

File: javascript/1903.js
var largestOddNumber = function (num) {
    for (let i = num.length - 1; 0 <= i; i--) {
        if (["1", "3", "5", "7", "9"].includes(num[i])) {
            return num.slice(0, i + 1)
        }
    } return ""
};

console.log(largestOddNumber("52"))